lib/api/core/plugins/pluginContext.js

```diff
@@ -277,6 +277,10 @@ function registerStrategy(Strategy, name, context, verify, options = {}) {
 
   try {
     this.passport.use(name, new Strategy(opts, verify.bind(context)));
+    
+    if (this.pluginsManager.registeredStrategies.indexOf(name) === -1) {
```

Comment:
if a strategy with the same name already exists, we should throw with an error and abort the plugin registration.
(and it should be done before call of `this.passport.use`)